jobsdb/jobsdb.go

```diff
@@ -1338,39 +1344,50 @@ func (jd *Handle) createDSInTx(tx *Tx, newDS dataSetT) error {
 		expire_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW());`, newDS.JobTable)); err != nil {
 		return err
 	}
-	if _, err = tx.ExecContext(ctx, fmt.Sprintf(`CREATE INDEX "idx_%[1]s_ws" ON %[1]q (workspace_id)`, newDS.JobTable)); err != nil {
-		return err
-	}
-	if _, err = tx.ExecContext(ctx, fmt.Sprintf(`CREATE INDEX "idx_%[1]s_cv" ON %[1]q (custom_val)`, newDS.JobTable)); err != nil {
-		return err
-	}
-	for _, param := range cacheParameterFilters {
-		if _, err = tx.ExecContext(ctx, fmt.Sprintf(`CREATE INDEX "idx_%[1]s_%[2]s" ON %[1]q USING BTREE ((parameters->>'%[2]s'))`, newDS.JobTable, param)); err != nil {
-			return err
-		}
-	}
-
-	if _, err = tx.ExecContext(ctx, fmt.Sprintf(`CREATE TABLE %q (
+	if _, err := tx.ExecContext(ctx, fmt.Sprintf(`CREATE TABLE %q (
 		id BIGSERIAL,
-		job_id BIGINT REFERENCES %q(job_id),
+		job_id BIGINT,
 		job_state VARCHAR(64),
 		attempt SMALLINT,
 		exec_time TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
 		retry_time TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
 		error_code VARCHAR(32),
 		error_response JSONB DEFAULT '{}'::JSONB,
 		parameters JSONB DEFAULT '{}'::JSONB,
-		PRIMARY KEY (job_id, job_state, id));`, newDS.JobStatusTable, newDS.JobTable)); err != nil {
+		PRIMARY KEY (job_id, job_state, id));`, newDS.JobStatusTable)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (jd *Handle) createDSIndicesInTx(ctx context.Context, tx *Tx, newDS dataSetT) error {
```

Comment:
Should we add more context around the error branches? Same for `createDSTablesInTx` as well.